app/suggestion/views.py

```diff
@@ -1,16 +1,115 @@
-from flask import render_template
+import pytz
+
+from datetime import datetime
+from flask import abort, flash, redirect, render_template, url_for
 from flask.ext.login import login_required
+from sqlalchemy.exc import IntegrityError
+
 from . import suggestion
-from ..models import Suggestion
+from .. import db
+from ..models import Resource, Suggestion
+from forms import SuggestionForm
 
 
 @suggestion.route('/')
 @login_required
 def index():
+    """View all suggestions in a list."""
     suggestions = Suggestion.query.all()
     return render_template('suggestion/index.html', suggestions=suggestions)
 
 
-@suggestion.route('/suggest-new')
-def suggest_new():
-    return render_template('suggestion/suggest_new.html')
+@suggestion.route('/unread')
+@login_required
+def unread():
+    """Returns the number of unread suggestions."""
+    num_unread = Suggestion.query.filter(
+        Suggestion.read == False  # noqa
+        ).count()
+    return "%d" % num_unread
+
+
+@suggestion.route('/toggle-read/<int:sugg_id>')
+@login_required
+def toggle_read(sugg_id):
+    """Toggles the readability of a given suggestion."""
+    suggestion = Suggestion.query.get(sugg_id)
+    if suggestion is None:
+        abort(404)
+    suggestion.read = not suggestion.read
+    db.session.add(suggestion)
+    try:
+        db.session.commit()
+    except:
+        db.session.rollback()
+        flash('Database error occurred. Please try again.', 'error')
+    return redirect(url_for('suggestion.index'))
+
+
+@suggestion.route('/delete/<int:sugg_id>')
+@login_required
+def delete(sugg_id):
+    """Delete a given suggestion."""
+    suggestion = Suggestion.query.get(sugg_id)
+    if suggestion is None:
+        abort(404)
+    db.session.delete(suggestion)
+    try:
+        db.session.commit()
+        flash('Suggestion successfully deleted.', 'success')
+    except:
+        db.session.rollback()
+        flash('Database error occurred. Please try again.', 'error')
+    return redirect(url_for('suggestion.index'))
+
+
+@suggestion.route('/create',  methods=['GET', 'POST'])
```

Comment:
could you combine the route handlers for new resource suggestion and edit suggestion? The only thing that's different is the presence/absence of resource_id, right?
So maybe either have no resource id for creation, or have a dummy resource id (if having no resource id will make Flask throw 404 instead of pattern matching to this route)